hive/envs/gym_env.py

```diff
@@ -54,19 +54,20 @@ def create_env_spec(self, env_name, **kwargs):
         )
 
     def reset(self):
-        observation = self._env.reset()
+        observation, _ = self._env.reset(seed=self._seed)
+        self._seed = None
```

Comment:
This line is a bit weird at first glance. And I think it makes the code a bit unreadable ([without opening the reset method in gym](https://github.com/openai/gym/blob/6a04d49722724677610e36c1f92908e72f51da0c/gym/core.py#L150)). I wonder if we could do better here. @dapatil211 Do you have any take on this thing?

Reply:
I agree it's not the best, but I think we have to do something like this because of gym's API change. They made the change because they don't want to force environments to have to support changing seeds in the middle of an episode, but I think maintaining the API where you separate out the seeding call is worthwhile, and this is the only way to do it. Let me know if you have a better idea.